File: src/main/scala/io/vertx/ext/asyncsql/mysql/impl/MysqlConnectionImpl.scala
package io.vertx.ext.asyncsql.mysql.impl

import com.github.mauricio.async.db.Connection
import io.vertx.core.Vertx
import io.vertx.ext.asyncsql.impl.BaseConnectionImpl
import io.vertx.ext.asyncsql.mysql.MysqlConnection

import scala.concurrent.Future

/**
 * @author <a href="http://www.campudus.com">Joern Bernhardt</a>.
 */
class MysqlConnectionImpl(val vertx: Vertx,
                               override protected val connection: Connection,
                               override protected val freeHandler: Connection => Future[_])
  extends BaseConnectionImpl with MysqlConnection with MysqlOverrides {

}

File: vertx-mysql-postgresql-client-scala/src/main/java/io/vertx/ext/asyncsql/impl/MySQLConnectionImpl.java
package io.vertx.ext.asyncsql.impl;

import com.github.mauricio.async.db.Connection;
import com.github.mauricio.async.db.QueryResult;
import com.github.mauricio.async.db.mysql.MySQLQueryResult;
import io.vertx.core.json.JsonArray;
import io.vertx.ext.asyncsql.impl.pool.AsyncConnectionPool;
import io.vertx.ext.sql.UpdateResult;
import scala.concurrent.ExecutionContext;

/**
 * @author <a href="http://www.campudus.com">Joern Bernhardt</a>.
 */
public class MySQLConnectionImpl extends AsyncSQLConnectionImpl {
  public MySQLConnectionImpl(Connection conn, AsyncConnectionPool pool, ExecutionContext ec) {
    super(conn, pool, ec);
  }

  @Override
  protected String getStartTransactionStatement() {
    return "BEGIN";
  }

  @Override
  protected UpdateResult queryResultToUpdateResult(QueryResult qr) {
    int affected = (int)qr.rowsAffected();
    MySQLQueryResult mySQLQueryResult = (MySQLQueryResult) qr;
    return new UpdateResult(affected, new JsonArray().add(mySQLQueryResult.lastInsertId()));
  }
}
